jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/actions/CopyArn.kt

```diff
@@ -3,14 +3,15 @@
 
 package software.aws.toolkits.jetbrains.core.explorer.actions
 
+import com.intellij.icons.AllIcons
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.ide.CopyPasteManager
 import software.aws.toolkits.resources.message
 import java.awt.datatransfer.StringSelection
 
-class CopyArn(private val arn: String) : AnAction(message("explorer.copy_arn")) {
+class CopyArn(private val arnProvider: () -> String) : AnAction(message("explorer.copy_arn"), null, AllIcons.Actions.Copy) {
```

Comment:
So crazy theory here...

```
class CopyArn : SingleResourceNodeAction<AwsExplorerResourceNode<*>>(message("explorer.copy_arn")) {
    override fun actionPerformed(selected: AwsExplorerResourceNode<*>, e: AnActionEvent) {
        CopyPasteManager.getInstance().setContents(StringSelection(selected.resourceArn()))
    }
}
```

We can then stick CopyArn into plugin.xml

Reply:
Only problem with that is that you then have to put it in the plugin.xml for each resource.

Reply:
Ya, but we were just complaining about how we don't like the ordering, so we would be more consistent and have more control

Reply:
What if we did the opposite - and created a new marker interface `DeleteableResource` that our menu creation can add automatically, we can then control the location of delete in the menu too.